engine/engine.go

```diff
@@ -948,3 +948,13 @@ func (bot *Engine) SetupExchanges() error {
 func (bot *Engine) WaitForInitialCurrencySync() error {
 	return bot.currencyPairSyncer.WaitForInitialSync()
 }
+
+// RegisterWebsocketRoutineHandlerInterceptor registers an externally defined
+// interceptor function for diverting and handling websocket notifications
+// across all enabled exchanges.
+func (bot *Engine) RegisterWebsocketRoutineHandlerInterceptor(fn Interceptor) error {
```

Comment:
Could also change the signature to `RegisterWebsocketRoutineHandlerInterceptor(fn Interceptor, interceptorOnly bool)`

So that way, in `(m *websocketRoutineManager) WebsocketDataReceiver(ws *stream.Websocket)`, you can do:
```go
			m.mu.RLock()
			if m.interceptor != nil {
				err = m.interceptor(ws.GetName(), data)
				if err != nil {
				     log.Errorf(...)
				}
				if m.interceptorOnly {
					m.mu.RUnlock()
					continue
				}
			}
			m.mu.RUnlock()
```